### ISSUE TITLE
Support newer event type

### ISSUE THREAD

**Issue by tribela** [halcy/Mastodon.py#234]
Mastodon 3.5 introduces `status.update` event.
Currently mastodon.py doesn't support it, And keep crashing on that event.

**Comment by arittner** [comment]
Hi!

I guess you mean this from the changelog:


> Add support for incoming edited posts (https://github.com/mastodon/mastodon/pull/16697, https://github.com/mastodon/mastodon/pull/17727, https://github.com/mastodon/mastodon/pull/17728, https://github.com/mastodon/mastodon/pull/17320, https://github.com/mastodon/mastodon/pull/17404, https://github.com/mastodon/mastodon/pull/17390, https://github.com/mastodon/mastodon/pull/17335, https://github.com/mastodon/mastodon/pull/17696, https://github.com/mastodon/mastodon/pull/17745, https://github.com/mastodon/mastodon/pull/17740, https://github.com/mastodon/mastodon/pull/17697, https://github.com/mastodon/mastodon/pull/17648, https://github.com/mastodon/mastodon/pull/17531, https://github.com/mastodon/mastodon/pull/17499, https://github.com/mastodon/mastodon/pull/17498, https://github.com/mastodon/mastodon/pull/17380, https://github.com/mastodon/mastodon/pull/17373, https://github.com/mastodon/mastodon/pull/17334, https://github.com/mastodon/mastodon/pull/17333, https://github.com/mastodon/mastodon/pull/17699, https://github.com/mastodon/mastodon/pull/17748)

>    Previous versions remain available for perusal and comparison
>    People who reblogged a post are notified when it's edited
>    New REST APIs:
>        `PUT /api/v1/statuses/:id`
>        `GET /api/v1/statuses/:id/history`
>        `GET /api/v1/statuses/:id/source`
>    New streaming API event:
>        `status.update`

And the crash is in the streaming API on receiving the update event?

**Comment by arittner** [comment]
Hello @tribela 

I guess the "crash" is a MastodonMalformedEventError from this function (in [streaming.py](https://github.com/halcy/Mastodon.py/blob/master/mastodon/streaming.py#L119)):

~~~python
    def _dispatch(self, event):
...
        handler_name = 'on_' + name
        try:
            handler = getattr(self, handler_name)
        except AttributeError as err:
            exception = MastodonMalformedEventError('Bad event type', name)
            self.on_abort(exception)
            six.raise_from(
               exception,
               err
            )
        else:
            handler(payload)
~~~

The `name` of the event-type is not supported. The Mastodon.py supports only `update` (which means a new status), `delete`, `notification` and `conversation`.

I'll try to support @halcy to implement it.

It may also be useful to configure Mastodon.py to ignore unknown events to avoid future problems with new events. With this option, a developer has the choice to use Mastodon.py with newer API events.

**Comment by arittner** [comment]
I checked the code, and it's typically possible to write listener-functions to manage well-known new events. Unfortunately, Mastodon API use now a dot inside the event name. And this is not supported with the callback-function lookup. 

So, we have two cases: 

1. Unknown new events breaks existing listeners. We have no handler to manage complete unknown messages. So every time the API comes with new events, a developer must enhance the listener. Otherwise, the app will break on fetching these unknown events.
2. Events with reserved characters (e.g. a `.` dot) cannot be handled, because this is not allowed:
   ~~~python
       def on_status.update(self, data):
           pass
   ~~~

For point 2 the solution is simple. The event-name can be patched to replace dots with underscores (we hope, we'll not get other special characters). The solution for the first point is just a little more complicated, but worth it if you want to prevent such current cases that Mastodon.py is not usable for a while when the API has changed.

I implemented both solutions and check it, if it's working. I'll drop a PR.

/cc @halcy
